lib/compress/zstd_compress.c

```diff
@@ -2626,6 +2657,166 @@ ZSTD_compressSequences_internal(seqStore_t* seqStorePtr,
     return op - ostart;
 }
 
+MEM_STATIC size_t ZSTD_buildEntropy_literal(const void* src, size_t srcSize,
+                                            const ZSTD_hufCTables_t* prevHuf,
+                                            ZSTD_hufCTables_t* nextHuf,
+                                            ZSTD_hufCTablesMetadata_t* hufMetadata)
+{
+    unsigned count[HUF_SYMBOLVALUE_MAX + 1];
+    unsigned maxSymbolValue = 255;
+    unsigned huffLog = 11;
+
+    DEBUGLOG(5, "ZSTD_buildEntropy_literal (srcSize=%zu)", srcSize);
+
+    /* Prepare nextEntropy assuming reusing the existing table */
+    memcpy(nextHuf, prevHuf, sizeof(*prevHuf));
+
+    /* small ? don't even attempt compression (speed opt) */
```

Comment:
This limit might not be the good one in this context.

The "small size" shortcut is a good idea when cumulating both the huffman header and huffman compressed symbols.

When the huffman header is essentially "free", because it's re-used from previous block (or from dictionary), the limit becomes much lower. In [this code](https://github.com/facebook/zstd/blob/dev/lib/compress/zstd_compress.c#L1957), the limit is down to `6` when huffman header is implicit.

Reply:
Currently, I am not considering set_repeat to simplify problem. I think I should add back the reduced limit check when I add the logic for set_repeat back in. What do you think?

Reply:
OK, so these are statistics for the superblock.
It makes sense, since there is no need (yet) to rebuild entropy tables between sub-blocks.
And, yes, in this case, your suggestion is completely reasonable.

Yet, since I did not understood it was statistics for the superblock,
it probably implies the naming can be improved a bit.